nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -356,6 +358,70 @@ impl<'a> Planner<'a> {
 
         Ok(())
     }
+
+    fn do_plan_cockroachdb_cluster_settings(&mut self) {
+        // Figure out what we should set the CockroachDB "preserve downgrade
+        // option" setting to based on the planning input.
+        //
+        // A given major version of CockroachDB (such as '22.2') is backward
+        // compatible with the storage format of the previous version of
+        // CockroachDB. This is shown by the `version` cluster setting, which
+        // displays the current storage format version. When `version` is
+        // '22.2', versions v22.2.x or v23.1.x can be used to run a node. This
+        // allows for rolling upgrades of nodes within the cluster.
+        //
+        // By default, when all nodes of a cluster are upgraded to a new major
+        // version, the upgrade is "auto-finalized"; `version` is changed to the
+        // new major version, and rolling back to a previous major version of
+        // CockroachDB is no longer possible.
+        //
+        // The `cluster.preserve_downgrade_option` cluster setting can be used
+        // to control this. This setting can only be set to the current value of
+        // the `version` cluster setting, and when it is set, CockroachDB will
+        // not perform auto-finalization. To perform finalization and finish
+        // the upgrade, a client must reset the preserve downgrade option.
+        // Finalization occurs in the background, and the preserve downgrade
+        // option cannot be set again until finalization completes.
+        //
+        // This table describes the logic we take here:
+        //
+        //            the `version` cluster setting
+        //            | | |    the `cluster.preserve_downgrade_option` setting
+        //            v v v    v v v
+        // +--------+--------+--------+--------------------------------------+
+        // | policy |  vers. | p.d.o. | Action                               |
+        // +--------+--------+--------+--------------------------------------+
+        // |  22.1  =  22.1  |  NULL  | Set p.d.o. to 22.1                   |
+        // |  22.1  =  22.1  |  22.1  | Do nothing (ideal state)             |
+        // +--------+--------+--------+--------------------------------------+
+        // |  22.2  ≠  22.1  |  22.1  | Reset p.d.o.                         |
+        // |  22.2  ≠ 22.1-9 |  NULL  | Finalizing -- do not change settings |
+        // +--------+--------+--------+--------------------------------------+
```

Comment:
> This also reflects the version of CockroachDB software that should be running.

Not quite, since in R10 we'll ship 22.2 but still have this policy be 22.1. I'll reword though.